data/StateMachine.j2119

```diff
@@ -28,7 +28,9 @@ A Pass State MAY have a field named "Result".
 A Fail State MUST NOT have a field named "InputPath".
 A Fail State MUST NOT have a field named "OutputPath".
 A Fail State MAY have a string field named "Cause".
+A Fail State MAY have a referencePath field named "CausePath".
```

Comment:
`referencePath` only covers _some_ of the functionality.

These fields _also_ allow Intrinsic Functions that return a string. [Here's the full list](https://github.com/aws/amazon-states-language-service/blob/84829de062bfc4c0bbb5f644ae49a4f2c3b8338a/src/validation/validateStates.ts#L33)

As is, the referencePath rule will fail valid state machines, for example:
```
{
  "StartAt": "FailState",
  "States": {
    "FailState": {
      "Type": "Fail",
      "ErrorPath": "States.Format('this is the error: {}, and this is the cause: {}', $.Error, $.Cause)",
      "CausePath": "States.UUID()"
    }
  }
}
```

Reply:
True, we'll have to just accept any field as is done for `ResultSelector`.